elastic/beats
# Fix bug in manual calculation of actual mem data

## What does this PR do?

This fixes an easily missed bug in the fallback code that populates `actual.free` in cases where the user is running an older kernel that doesn't provide a `MemAvailable` counter. This also adds a few tests to make sure it works.

## Why is it important?

This is a bug, and under certain conditions memory can be calculated incorrectly.

## Checklist


## How to test this PR locally

- Run an on older kernel where there's no `MemAvailable` field in `/proc/meminfo`
- Make sure `system.memory.actual.free` is correct.
